packages/fast-stable-stringify/src/index.ts

```diff
@@ -0,0 +1,77 @@
+const objToString = Object.prototype.toString;
+const objKeys =
+    Object.keys ||
+    function (obj) {
+        const keys = [];
+        for (const name in obj) {
+            keys.push(name);
+        }
+        return keys;
+    };
+
+function stringify(val, isArrayProp) {
+    let i, max, str, keys, key, propVal, toStr;
+    if (val === true) {
+        return 'true';
+    }
+    if (val === false) {
+        return 'false';
+    }
+    switch (typeof val) {
+        case 'object':
+            if (val === null) {
+                return null;
+            } else if (val.toJSON && typeof val.toJSON === 'function') {
+                return stringify(val.toJSON(), isArrayProp);
+            } else {
+                toStr = objToString.call(val);
+                if (toStr === '[object Array]') {
+                    str = '[';
+                    max = val.length - 1;
+                    for (i = 0; i < max; i++) {
+                        str += stringify(val[i], true) + ',';
+                    }
+                    if (max > -1) {
+                        str += stringify(val[i], true);
+                    }
+                    return str + ']';
+                } else if (toStr === '[object Object]') {
+                    // only object is left
+                    keys = objKeys(val).sort();
+                    max = keys.length;
+                    str = '';
+                    i = 0;
+                    while (i < max) {
+                        key = keys[i];
+                        propVal = stringify(val[key], false);
+                        if (propVal !== undefined) {
+                            if (str) {
+                                str += ',';
+                            }
+                            str += JSON.stringify(key) + ':' + propVal;
```

Comment:
Now that there is definitely _no_ expectation that the hash can be used with `JSON.parse`, I wonder if we can eliminate this `JSON.stringify` and just output `String(key).replace(/:/g, '\:')`?

Reply:
Although, I just read this… https://github.com/nickyout/fast-stable-stringify?tab=readme-ov-file#note-about-string-escaping

Reply:
To be clear: this would just be to reduce the size of the hash string.

Reply:
Can you give me the input and output of a sample json?

Reply:
I'm actually a little hesitant to make this change now, since we no longer have _performance_ benchmarks here. I would not want to make it slower by making the output more compact.

I was talking about this:

```
s({"cool:beans": "bar"}) // {cool\:beans:"bar"}
```